### ISSUE TITLE
PMAX8000 Firmware Update Breaks get_replication_info()

### ISSUE THREAD

**Issue by prolly-brent** [dell/PyU4V#142]
EMC recently updated our PMAX8000 to ucode level 5978.711.711.  Immediately after the update, the "get_replication_info() API module began erroring with:

Traceback (most recent call last):
  File "./check_pmax_cache_rdp.py", line 76, in <module>
    result = conn.replication.get_replication_info()
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/PyU4V/replication.py", line 78, in get_replication_info
    resource_level_id=self.array_id)
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/PyU4V/common.py", line 376, in get_resource
    target_uri, resource_type, kwargs.get('params'))
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/PyU4V/common.py", line 349, in get_request
    self.check_status_code_success(operation, sc, message)
  File "/opt/rh/rh-python36/root/usr/lib/python3.6/site-packages/PyU4V/common.py", line 183, in check_status_code_success
    data=exception_message)
PyU4V.utils.exception.VolumeBackendAPIException: Bad or unexpected response from the storage volume backend API: Error GET symmetrix. The status code received is 500 and the message is None.

We have a case/ticket opened with EMC and they have done some array log analysis, but do not see any issue on their end.  Please let us know if there is a potential problem with the PyU4V modules.

Note: We are running version 9.2.1.5 of PyU4V under python version 3.6.9.

Thanks,
Brent

**Comment by rawstorage** [comment]
Hi Brent, are you able to connect to the Unisphere instance?,  I know as part of the upgrade the embedded Unipshere instance will be inaccessible for a few minutes but once it's back PyU4V should be able to communicate.  If you can add the SR number I should be able to get your contact details from that and reach out direct.  

edit:: seems like the issue is with the endpoint being used by replication.get_replication_info()  there is a bug in Unipshere 9.2.1 that has affected the return on this call.    It's fixed in Unisphere 9.2.3.4 available here https://www.dell.com/support/home/en-uk/product-support/product/unisphere-powermax/drivers, if you are running embedded Unisphere you will need schedule this upgrade.

**Comment by prolly-brent** [comment]
Hi Paul,

Thanks for getting back with me.  We did verify with EMC through direct REST API calls that there is an issue with the /92 replication endpoint.  It works fine in /91.  We are currently running version 9.2.2.2 of Unisphere and according to your response we may need to update.  I have passed that information along to EMC support who we are still waiting on for a solution.  The SR# is 129676625.

Thanks,
Brent
